### PR TITLE
Add dns-util module for code review

### PR DESCRIPTION
**Pull request by srsudar**

### CODE REVIEW

File: chromeapp/test/scripts/dnssd/dns-util-test.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
+'use strict';
```
Comment (mdwelsh) [inline chromeapp/test/scripts/dnssd/dns-util-test.js:1]: Consider adding tests to cover the various error cases - you seem to mostly be testing the sane cases here. Also suggest testing for domains with non-ASCII Unicode characters. You can do this in a follow-up CL if you like.

**Comment by mdwelsh** [comment]
LGTM but see note about better tests.

File: chromeapp/test/scripts/dnssd/dns-util-test.js (hunk 1 above)
Comment (srsudar) [inline chromeapp/test/scripts/dnssd/dns-util-test.js:1]: I opened Issue #14 for unicode.

I also added tests for the short and long extremes of valid URLs, as well as asserted that all but a sanity check `throw` statement does indeed throw.
